### ISSUE TITLE
Scope description overflows the allocated width in scope list item in New API Resource modal

### ISSUE THREAD

**Issue by pavinduLakshan** [wso2/product-is#20599]
**Describe the issue:**

<img width="1207" alt="Screenshot 2024-06-27 at 13 26 59" src="https://github.com/wso2/product-is/assets/41533942/055c2721-7ec9-40ff-b1a1-ad4ed3efbbfa">

**How to reproduce:**

N/A

**Expected behavior:**

Scope description should not be overflown.

**Environment information** (_Please complete the following information; remove  any unnecessary fields_) **:**
 - Product Version: IS 7.0.0
 - OS: [e.g., Windows, Linux, Mac]
 - Database: [e.g., MySQL, H2]
 - Userstore: [e.g., LDAP, JDBC]

**Comment by Iduranga-Uwanpriya** [comment]
can i work on this isuue

**Comment by sagara-gunathunga** [comment]
Great to see your interest @Iduranga-Uwanpriya! Would you mind completing  1 or 2 issues from your other assigned issues before picking this up?

**Comment by Iduranga-Uwanpriya** [comment]
Yes Sure Sir I will Complete My First And get back to this soon

**Comment by yasandu0505** [comment]
I can work on this @sagara-gunathunga

**Comment by d3varaja** [comment]
Hey @sagara-gunathunga, I see there’s already some interest in this issue. If it’s still open for contribution, I’d be glad to work on it. Could you let me know if I can proceed?

**Comment by sagara-gunathunga** [comment]
@yasandu0505 Since you don’t have any other issues assigned on our board, I’m assigning this one to you. Please provide regular updates here so we know it’s actively being worked on.

**Comment by yasandu0505** [comment]
> @yasandu0505 Since you don’t have any other issues assigned on our board, I’m assigning this one to you. Please provide regular updates here so we know it’s actively being worked on.

Sure i will update you guys. Thank You

**Comment by s-vamshi** [comment]
hey @yasandu0505, I noticed this issue hasn’t been updated recently, if you’re currently working on other tasks, I’d be happy to take on this issue. Please let me know if that’s alright.

@sagara-gunathunga 
there are of total 2 issues assigned to me under wso2 project where for both of them i have created pr.

**Comment by sagara-gunathunga** [comment]
Due to a long period of inactivity, we have unassigned this issue. It is now available for other contributors to pick up.

@s-vamshi Are you still interested in taking this issue?

**Comment by s-vamshi** [comment]
@sagara-gunathunga yes i am interested can you please assign this to me?

**Comment by s-vamshi** [comment]
@pavinduLakshan can you please confirm if the description should be wrapped?

**Comment by pavinduLakshan** [comment]
> [@pavinduLakshan](https://github.com/pavinduLakshan) can you please confirm if the description should be wrapped?

Yes. This has already been fixed in the scopes tab in the API resource edit view. Let's make the UX here consistent with that.

<img width="1124" height="399" alt="Image" src="https://github.com/user-attachments/assets/dbae63e8-1dbe-4c8f-914a-22a4751c1235" />

**Comment by s-vamshi** [comment]
@pavinduLakshan i am not able insert so much text as you did in screenshot
At max i am only able to add 300 chars

**Comment by pavinduLakshan** [comment]
> [@pavinduLakshan](https://github.com/pavinduLakshan) i am not able insert so much text as you did in screenshot At max i am only able to add 300 chars

That’s okay. I temporarily added this long text through the browser devtools just to demonstrate the intended final UX. Let’s optimize the UX for the 300-character limit.

**Comment by s-vamshi** [comment]
@pavinduLakshan this issue only exists when we have description without any spaces between words(happens both while creating api and in list api)

<img width="1524" height="731" alt="Image" src="https://github.com/user-attachments/assets/673f7a59-a630-45e9-b55a-42d94a3ef6bb" />

and also this issue is exists while creating new mcp server  and  in scope of an mcp server also

<img width="1503" height="699" alt="Image" src="https://github.com/user-attachments/assets/b9e56599-fbe9-4140-8455-e5a68a14789a" />

**Comment by s-vamshi** [comment]
@pavinduLakshan and also if i try to add scope from scopes section by clicking on add scopes with description length more than 300 then i am getting error by it doesn't clearly mention about this max 300 length

<img width="655" height="295" alt="Image" src="https://github.com/user-attachments/assets/00a322df-2b1f-4859-83a2-229f525714a2" />

<img width="1918" height="30" alt="Image" src="https://github.com/user-attachments/assets/e846303b-d8a7-4976-9469-254febb27a89" />

<img width="810" height="140" alt="Image" src="https://github.com/user-attachments/assets/45c788e1-3029-40cc-8704-f4622ca12de5" />

**Comment by pavinduLakshan** [comment]
> this issue only exists when we have description without any spaces between words

Shall we check why it only happens for long text without spaces. I guess a CSS property for text wrapping could be causing this inconsistency.

> and also if i try to add scope from scopes section by clicking on add scopes with description length more than 300 then i am getting error by it doesn't clearly mention about this max 300 length

Yes, agreed that the UX should be improved here. We will handle it in a separate effort. Thanks for the report though!

**Comment by s-vamshi** [comment]
>Shall we check why it only happens for long text without spaces. I guess a CSS property for text wrapping could be causing this inconsistency.

Yes will look into this

**Comment by s-vamshi** [comment]
@pavinduLakshan setting overflow-wrap to anywhere is working though

<img width="1663" height="534" alt="Image" src="https://github.com/user-attachments/assets/dfb2dc08-ea1c-4187-93fb-ea84d10f05c0" />

can this css be used?

**Comment by sagara-gunathunga** [comment]
@s-vamshi Congratulations on completing this issue! Unless you’ve already done so, please complete the form for this issue as mentioned on the WSO2 Hacktoberfest webpage: https://wso2.com/hacktoberfest/
